### PR TITLE
Added comments to stable shaders

### PR DESCRIPTION
**Pull request by jarisiru**

### CODE REVIEW

File: shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag [hunk 1]
```diff
@@ -23,20 +23,39 @@ layout(location = 0) out vec4 _GLF_color;
...
         } else {
             if (mod(float(i), round(thirty_two)) <= 0.01) {
-            // This should never get executed, because the previous if condition would get triggered.
+            // This should never get executed, because the 
+            // previous if condition would get triggered.
                 result += 100.0;
             }
+            // if i % 32 isn't zero, result is untouched
         }
+        // early out based on gl_FragCoord:
```
Comment (afd) [inline shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag:58]: early out -> early exit

File: shaders/src/main/glsl/samples/320es/stable_maze.frag [hunk 2]
```diff
@@ -22,24 +22,53 @@ precision highp int;
...
+such be found, the walker is teleported there.
+
+If there are directions to walk to, steps are taken
+with a little bit of randomness based on how many
+legal moves have been possible and how many directions
+may be taken. All of this is done with integer math,
+so the result is deterministic.
+
+(For different kinds of mazes, just change the
+starting position of the walker)
```
Comment (afd) [inline shaders/src/main/glsl/samples/320es/stable_maze.frag:45]: add period after "walker" before ")"

File: shaders/src/main/glsl/samples/320es/stable_maze.frag [hunk 3]
```diff
@@ -100,11 +140,15 @@ void main() {
...
     if (map[ipos.y * 16 + ipos.x] == 1) {
+      // Early out: our pixel is in a wall
       _GLF_color = vec4(1.0, 1.0, 1.0, 1.0);
       return;
     }
   }
   while (canwalk);
-
+  
+  // If we manage to get here, there's no legal moves left,
```
Comment (afd) [inline shaders/src/main/glsl/samples/320es/stable_maze.frag:150]: there's -> there are

File: shaders/src/main/glsl/samples/320es/stable_maze.frag [hunk 4]
```diff
@@ -100,11 +140,15 @@ void main() {
...
+  // and our pixel is in a non-wall tile.
   _GLF_color = vec4(0.0, 0.0, 0.0, 1.0);
 }
+
```
Comment (afd) [inline shaders/src/main/glsl/samples/320es/stable_maze.frag:154]: Similar comment to the above if this is an extra line (as opposed to an EOF newline that was missing).

File: shaders/src/main/glsl/samples/320es/stable_mergesort.frag [hunk 5]
```diff
@@ -25,6 +25,25 @@ uniform vec2 injectionSwitch;
 
 uniform vec2 resolution;
 
+/*
+This shader populates data array with reversed values,
+performs merge sort to get the data to the correct order,
+and then draws an image based on the sorted values and
+the screen coordinates.
+
+The sorting operation is purely integer math, and the
```
Comment (afd) [inline shaders/src/main/glsl/samples/320es/stable_mergesort.frag:34]: "is purely integer math" -> "uses purely integer math"

File: shaders/src/main/glsl/samples/320es/stable_mergesort.frag [hunk 6]
```diff
@@ -25,6 +25,25 @@ uniform vec2 injectionSwitch;
...
+image drawing uses only linear operations (division, 
+addition) making it numerically stable.
+
+Populating the input data uses an injection switch 
+uniform to avoid the shader compiler from optimizing
+the initialization.
```
Comment (afd) [inline shaders/src/main/glsl/samples/320es/stable_mergesort.frag:40]: I would rephase to say that it uses a uniform named injectionSwitch that is guaranteed to be populated with the value (0.0, 1.0) at run-time.

File: shaders/src/main/glsl/samples/320es/stable_mergesort.frag [hunk 7]
```diff
@@ -83,50 +117,35 @@ void mergeSort()
 
 void main()
 {
+ // Note the use of injectionSwitch uniform to avoid compiler
+ // optimization of the population of initial data[] values.
```
Comment (afd) [inline shaders/src/main/glsl/samples/320es/stable_mergesort.frag:121]: I would remind the reader that injectionSwitch has the value (0.0, 1.0)

File: shaders/src/main/glsl/samples/320es/stable_quicksort.frag [hunk 8]
```diff
@@ -23,6 +23,19 @@ layout(location = 0) out vec4 _GLF_color;
...
 uniform vec2 resolution;
 
+/*
+This shader creates an array of 10 items, populates it
+with reversed values, and performs a quicksort to put
+the values in correct order. The sorted values are then
+used to produce a colorful image, which will display
+incorrectly if the sorting failed.
+
+Sorting itself is purely integer operation, while the
```
Comment (afd) [inline shaders/src/main/glsl/samples/320es/stable_quicksort.frag:33]: -> Sorting itself uses purely integer operations

File: shaders/src/main/glsl/samples/320es/stable_triangle.frag [hunk 9]
```diff
@@ -22,20 +22,34 @@ precision highp int;
 layout(location = 0) out vec4 _GLF_color;
 uniform vec2 resolution;
 
+
+/*
+This shader rasterizes a triangle in perhaps the most
+inefficent way possible apart from raytracing, by 
+checking for each pixel whether we're inside a triangle.
+
+The point in triangle code was based on
```
Comment (afd) [inline shaders/src/main/glsl/samples/320es/stable_triangle.frag:31]: "point in triangle" -> "pointInTriangle"

File: shaders/src/main/glsl/samples/320es/stable_bifurcation.frag [hunk 10]
```diff
@@ -40,15 +40,50 @@ const vec4 pal[16] = vec4[16](
...
+  
+/*
+This shader is basically a fancy random number generator
+using a simple bifurcation algorithm. All the math is 
+done using integers, so as long as the integer width is
+sufficient (which "precision highp int" should provide),
+the math itself is deterministic.
+
+The randomness comes from the unstability of the
+algorithm as the number of iterations increase.
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_bifurcation.frag]: done

File: shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag [hunk 11]
```diff
@@ -19,9 +19,42 @@
 precision highp float;
 precision highp int;
 
+/*
+This shader implements binary search tree using an array
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag]: done

File: shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag [hunk 12]
```diff
@@ -19,9 +19,42 @@
...
+data structure. The elements of tree are kept in the array
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag]: done

File: shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag [hunk 13]
```diff
@@ -19,9 +19,42 @@
 precision highp float;
 precision highp int;
 
+/*
+This shader implements binary search tree using an array
+data structure. The elements of tree are kept in the array 
+that contains a list of BST object holding indices of left
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag]: done

File: shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag [hunk 14]
```diff
@@ -19,9 +19,42 @@
 precision highp float;
 precision highp int;
 
+/*
+This shader implements binary search tree using an array
+data structure. The elements of tree are kept in the array 
+that contains a list of BST object holding indices of left 
+and right subtree in the array.
+
+- Tree representation of the number used in this shader:
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag]: done

File: shaders/src/main/glsl/samples/320es/stable_mergesort.frag (hunk 5 above)
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_mergesort.frag]: done

File: shaders/src/main/glsl/samples/320es/stable_mergesort.frag (hunk 6 above)
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_mergesort.frag]: Done

File: shaders/src/main/glsl/samples/320es/stable_mergesort.frag (hunk 7 above)
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_mergesort.frag]: Done

File: shaders/src/main/glsl/samples/320es/stable_quicksort.frag (hunk 8 above)
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_quicksort.frag]: Done

File: shaders/src/main/glsl/samples/320es/stable_triangle.frag (hunk 9 above)
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_triangle.frag]: Done

File: shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag [hunk 15]
```diff
@@ -19,9 +19,42 @@
...
+           9
+        /    \
+       5      12
+     /  \      \
+    2   7      15
+       / \    /  \
+      6  8   13  17
+- Array representation:
+  [9, 5, 12, 15, 7, 8, 2, 6, 17, 13]
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_binarysearch_tree.frag:37]: I think that shows the exact value of the internal representation of the array.

File: shaders/src/main/glsl/samples/320es/stable_bubblesort_flag.frag [hunk 16]
```diff
@@ -25,47 +25,68 @@ uniform vec2 injectionSwitch;
...
 {
     float data[10];
-    for(
-        int i = 0;
-        i < 10;
-        i ++
-    )
+    // Populate the data array in a reversed order.
+    // Note the use of injectionSwitch to avoid compiler
+    // optimizations.
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_bubblesort_flag.frag]: Done

File: shaders/src/main/glsl/samples/320es/stable_collatz.frag [hunk 17]
```diff
@@ -56,8 +79,34 @@ int collatz(int v) {
...
+ 10 124  31  31  31  80  18  31  31  31  18  18  93  93  18  44  18
+ 44 106 106 106  44  13 119 119 119  26  26  26 119  26  18  26  39
+ 26  26  88  88  13  39  39  39  13  13 101 101 114  26 114  52 114 
+114  70  70  21  52  13  13  34  34  34 127  21  83  21 127  34  34
+ 34  52  21  21  96  96  21  21  47  47 109  47 109  65 109 109  47 
+ 47   8 122 122 122  29  29  29  78  29 122  29  21  29  29  42  42
+ 16  29  91  91  16  16  42  42  16  42  16  60 104 104 104  42 
+*/  
 }
+
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_collatz.frag]: It's the missing newline.

File: shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag [hunk 18]
```diff
@@ -23,20 +23,39 @@ layout(location = 0) out vec4 _GLF_color;
 
 uniform vec2 resolution;
 
+/*
+This shader produces a grid of colored 32x32 pixel
+squares in a rather roundabout way. 
+
+While floating point numbers are used, the values
+are largely integer. The resulting image uses
+multiplication and normalize.
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag]: Rephrased. And then rephrased again.

File: shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag [hunk 19]
```diff
@@ -23,20 +23,39 @@ layout(location = 0) out vec4 _GLF_color;
...
+*/
+
+/*
+For every 32 values in limit, returns the values:
+-0.500 -0.100 0.300 0.700 1.100 1.500 1.900 2.300
+(i.e, for 0..31 returns -0.5, for 32..63 returns -0.1, etc)
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag]: done. I agree, it was a mess.

File: shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag [hunk 20]
```diff
@@ -23,20 +23,39 @@ layout(location = 0) out vec4 _GLF_color;
...
         if ((i % 32) == 0) {
-            // Avoid computing e.g. mod(float(32), round(32.0)), which could be sensitive to round-off if mutated.
+            // Avoid computing e.g. mod(float(32), round(32.0)), 
+            // which could be sensitive to round-off if mutated.
             result += 0.4;
         } else {
             if (mod(float(i), round(thirty_two)) <= 0.01) {
-            // This should never get executed, because the previous if condition would get triggered.
+            // This should never get executed, because the 
+            // previous if condition would get triggered.
```
[the remark this answers is not in the document]
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag]: Done

File: shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag (hunk 1 above)
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_colorgrid_modulo.frag]: Done

File: shaders/src/main/glsl/samples/320es/stable_maze.frag (hunk 2 above)
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_maze.frag]: Done

File: shaders/src/main/glsl/samples/320es/stable_maze.frag (hunk 3 above)
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_maze.frag]: Done

File: shaders/src/main/glsl/samples/320es/stable_maze.frag (hunk 4 above)
Comment (jarisiru) [reply shaders/src/main/glsl/samples/320es/stable_maze.frag]: And similarly, it was the missing newline.

**Pull request merged**